Lib/json/tool.py

```diff
@@ -55,7 +55,7 @@ def main():
             infile = open(options.infile, encoding='utf-8')
         try:
             if options.json_lines:
-                objs = (json.loads(line) for line in infile)
+                objs = tuple(json.loads(line) for line in infile)
```

Comment:
This has significant memory implications for large `.jsonl` files.  It might be better to instead move the `finally` clause either to the outer `try` or just after the `with outfile` block, but that does have implications for the `infile == outfile` case.

Reply:
Hmm, yeah, moving it gives a blank file when `infile == outfile`...

Reply:
Or we could load all the lines into a list. This uses more memory than before, but less than the tuple as we're only deserialising each line when consumed from the generator:
```suggestion
                lines = infile.readlines()
                objs = (json.loads(line) for line in lines)
```

I don't think it's worth adding much extra complexity for this CLI, as if someone needs to process extremely large `.jsonl` files they can write their own code to do exactly what they need.